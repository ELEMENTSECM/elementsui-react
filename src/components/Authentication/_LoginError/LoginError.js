import * as React from 'react';
import PropTypes from 'prop-types';
import { styles } from './LoginError.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';
import { FormattedMessage } from 'react-intl';
import MessageBar from '../../Surfaces/MessageBar';

export class LoginError extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			visible: this.props.type
		};
		this.classNames = classNamesFunction()(styles, this.props);
	}

	toggleError = () => {
		this.setState(ps => ({ ...ps, visible: false }));
	};

	componentWillReceiveProps(newProps) {
		this.setState(ps => ({ ...ps, visible: newProps.type }));
	}

	render() {
		const { root } = this.classNames;
		const { children, type } = this.props;
		return (
			<div id={this.props.id} className={root}>
				{this.state.visible ? (
					<MessageBar
						messageBarType="error"
						isMultiline={false}
						onDismiss={() => this.toggleError()}
						dismissButtonAriaLabel="Close"
						truncated={true}
						overflowButtonAriaLabel="Overflow">
						<b>
							<FormattedMessage id={type} />.
						</b>
						{` ${children || ''}`}
					</MessageBar>
				) : (
					''
				)}
			</div>
		);
	}
}

LoginError.propTypes = {
	/** HTML id tag of the root element */
	id: PropTypes.string,
	/** Error type */
	type: PropTypes.oneOf(['backgroundLoginFailed', 'noModuleLicense', 'noModule']).isRequired
};

export default styled(customizable('LoginError', ['theme'])(LoginError), styles);
