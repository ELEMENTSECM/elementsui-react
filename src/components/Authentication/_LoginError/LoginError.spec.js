import * as React from 'react';
import { shallow } from 'enzyme';
import LoginError from './LoginError';

describe('LoginError', () => {
	test('displays correctly', () => {
		const wrapper = shallow(
			<LoginError id="testLoginError" type="noModule">
				Some error text
			</LoginError>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
