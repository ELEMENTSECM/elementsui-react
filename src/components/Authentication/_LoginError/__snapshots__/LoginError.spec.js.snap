// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginError displays correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledCustomizedLoginError
    id="testLoginError"
    type="noModule"
  >
    Some error text
  </StyledCustomizedLoginError>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Some error text",
      "id": "testLoginError",
      "styles": [Function],
      "type": "noModule",
    },
    "ref": null,
    "rendered": "Some error text",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Some error text",
        "id": "testLoginError",
        "styles": [Function],
        "type": "noModule",
      },
      "ref": null,
      "rendered": "Some error text",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
